### PR TITLE
fix(openapi): Standardise API naming scheme for V2

### PR DESCRIPTION
**Pull request by AnuthaDev**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [x] Documentation
- [ ] CI/CD

## Description

Use kebab-case for all API endpoints in V2

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

This is mostly a documentation PR, doesn't require any additional test cases.

Verified that existing cypress tests are passing

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/openapi/src/routes/payment_method.rs [hunk 1]
```diff
@@ -192,3 +197,171 @@ pub async fn payment_method_delete_api() {}
     security(("ephemeral_key" = []))
 )]
 pub async fn default_payment_method_set_api() {}
+
+/// Payment Method - Create Intent
+///
+/// TODO: Description Here
```
Comment (AnuthaDev) [inline crates/openapi/src/routes/payment_method.rs:203]: Please suggest the description here

File: crates/openapi/src/routes/payment_method.rs [hunk 2]
```diff
@@ -192,3 +197,171 @@ pub async fn payment_method_delete_api() {}
...
+    tag = "Payment Methods",
+    operation_id = "Create Payment Method Intent",
+    security(("api_key" = []))
+)]
+#[cfg(feature = "v2")]
+pub async fn create_payment_method_intent_api() {}
+
+/// Payment Method - Confirm Intent
+///
+/// TODO: Description Here
```
Comment (AnuthaDev) [inline crates/openapi/src/routes/payment_method.rs:224]: Please suggest a description here as well

File: crates/openapi/src/routes/api_keys.rs [hunk 3]
```diff
@@ -64,7 +64,7 @@ pub async fn api_key_retrieve() {}
 /// Retrieve information about the specified API Key.
 #[utoipa::path(
     get,
-    path = "/v2/api_keys/{key_id}",
+    path = "/v2/api-keys/{key_id}",
```
[the remark this answers is not in the document]
Comment (AnuthaDev) [reply crates/openapi/src/routes/api_keys.rs]: As mentioned in #6496 we want to use `id`

File: api-reference-v2/api-reference/merchant-account/profile--list.mdx [hunk 4]
```diff
@@ -1,3 +1,3 @@
 ---
-openapi: get /v2/merchant_accounts/{account_id}/profiles
+openapi: get /v2/merchant-accounts/{account_id}/profiles
```
[the remark this answers is not in the document]
Comment (AnuthaDev) [reply api-reference-v2/api-reference/merchant-account/profile--list.mdx]: the `profile--list` one seems to have been added accidentally in #6143 . Removed

File: crates/api_models/src/payment_methods.rs [hunk 5]
```diff
@@ -709,7 +710,7 @@ pub struct PaymentMethodResponse {
 #[derive(Debug, serde::Deserialize, serde::Serialize, ToSchema, Clone)]
 pub struct PaymentMethodResponse {
     /// Unique identifier for a merchant
-    #[schema(example = "merchant_1671528864")]
+    #[schema(value_type = Option<String>, example = "merchant_1671528864")]
```
Comment (jarnura) [inline crates/api_models/src/payment_methods.rs:713]:
```suggestion
    #[schema(value_type = String, example = "merchant_1671528864")]
```

File: crates/openapi/src/routes/payment_method.rs (hunk 1 above)
Comment (jarnura) [reply crates/openapi/src/routes/payment_method.rs:203]: **Creates a payment method for customer with billing information and other metadata.**

File: crates/openapi/src/routes/payment_method.rs (hunk 2 above)
Comment (jarnura) [reply crates/openapi/src/routes/payment_method.rs:224]: Update a payment method with customer's payment method related information.

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — COMMENTED** [review]
Any changes touches v1?

**Comment by AnuthaDev** [comment]
> Any changes touches v1?

Added description for fields in `OrganizationResponse`

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by pixincreate — DISMISSED** [review]
Cypress changes, LGTM

**Review by pixincreate — APPROVED** [review]

**Review by jarnura — DISMISSED** [review]

File: crates/openapi/src/routes/payment_method.rs [hunk 6]
```diff
@@ -192,3 +197,171 @@ pub async fn payment_method_delete_api() {}
...
+    tag = "Payment Methods",
+    operation_id = "Retrieve Payment Method",
+    security(("api_key" = []))
+)]
+#[cfg(feature = "v2")]
+pub async fn payment_method_retrieve_api() {}
+
+/// Payment Method - Update
+///
+/// Update an existing payment method of a customer. This API is useful for use cases such as updating the card number for expired cards to prevent discontinuity in recurring payments.
```
Comment (Sarthak1799) [inline crates/openapi/src/routes/payment_method.rs:286]: Are we sure we allow this card number update as of now?

**Review by SanchithHegde — DISMISSED** [review]

**Review by pixincreate — APPROVED** [review]

**Review by Sarthak1799 — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**